pkg/pillar/cmd/zedagent/hardwareinfo.go

```diff
@@ -57,7 +57,7 @@ func PublishHardwareInfoToZedCloud(ctx *zedagentContext) {
 	// Get information about disks
 	disksInfo, err := hardware.ReadSMARTinfoForDisks()
 	if err != nil {
-		log.Fatal("PublishHardwareInfoToZedCloud get information about disks failed. Error: ", err)
+		log.Error("PublishHardwareInfoToZedCloud get information about disks failed. Error: ", err)
```

Comment:
That does not warranty a Fatal crash. In fact we need system to be up and running to debug.

Reply:
@zedi-pramodh unfortunately with your approach it would have taken boths to discover and root cause this issue, since it would have been hidden. We can talk about the philosophy in a different forum than this PR.

My understanding is that the actual failure (which you did not include in the description) is "PublishHardwareInfoToZedCloud proto marshaling error: string field contains invalid UTF-8".
That is telling us that something defined as a string is being filled with something which is not a valid string. Reading the code I see two potential sources:
	diskInfo.ModelNumber = devIdentify.ModelNumber() // XXX clean
	diskInfo.SerialNumber = devIdentify.SerialNumber() // XXX clean

We should remove any non-printable characters from those, for example by replacing them with a '.' or a space.

Reply:
I agree my philosophy has both pros and cons. We can discuss later. But just curious on how did you narrow down to only ModelNumber and SerialNumber ? It could be WWN too. I think we need to provide a patch to dump all the values to see which one is bad.

Reply:
We have logic for that already in pkg/pillar/hardware/model.go massageCompatible() which replaces NUL with a period and then filters out non-printable characters. We should just apply that to ModelNumber and SerialNumber.

Reply:
Amended the commit.